bin/agd

```diff
@@ -0,0 +1,172 @@
+#! /bin/bash
+# bin/agd - A build wrapper to bootstrap the Agoric daemon
+#
+# This uses SKIP_DOWNLOAD=false or the Cosmovisor $DAEMON_HOME and
+# $DAEMON_ALLOW_DOWNLOAD_BINARIES variables to enable the automatic downloading
+# of Golang and NodeJS versions needed to build and run the current version of
+# the SDK.
+set -ueo pipefail
+
+GVM_URL=${GVM_URL-https://github.com/devnw/gvm/releases/download/latest/gvm}
+NVM_GIT_REPO=${NVM_GIT_REPO-https://github.com/nvm-sh/nvm.git}
+
+STAMPS=node_modules/.cache/agoric
+
+real0=$(readlink "${BASH_SOURCE[0]}" || echo "${BASH_SOURCE[0]}")
+thisdir=$(cd "$(dirname -- "$real0")" >/dev/null && pwd -P)
+
+# shellcheck disable=SC1091
+source "$thisdir/../repoconfig.sh"
+
+if test "${1-''}" = build; then
```

Comment:
I added more help text to the top of the file.  Determining arg parsing errors is tricky, because we accept all the options that `agd` does, but now I've made that clearer.